### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by serdg0**
Friendship V1

### CODE REVIEW

File: app/models/friendship.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Friendship < ApplicationRecord
+  belongs_to :requestor, class_name: 'User'
+  belongs_to :requested, class_name: 'User'
+
+  def user_requestor
```
Comment (mosaaleb) [inline app/models/friendship.rb:7]: - [x] `user_requestor` and `user_requested` method are kind of redundant. If you want to get the requestor of a specific friendship, you can use the `belongs_to` association above:
```ruby
friendship_instance.requestor
friendship_instance.requested
```

**Review by mosaaleb — CHANGES_REQUESTED** [review]
### Well done @serdg0 @MephistoDevelop :clap:

#### You did a great job handling all the friendship associations. Yet there are some requirements needs handling before moving on.

- [x] **`[OPTIONAL]`** sender sent a friend request to receiver. reciever can't send a friend request to sender.

We can achieve this by adding interchangeable validation to check if the [user_id, friend_id] or [friend_id, user_id] exist in the friendship table. [please take a look at this example](https://stackoverflow.com/questions/29123620/rails-validate-unique-friendship-relationships)

- [x] **`[OPTIONAL]`** Previous step adds model validations to friendships uniqueness. I would suggest to add interchangeable DB validations. You need to know first why DB validations are important [why](https://thoughtbot.com/blog/the-perils-of-uniqueness-validations). 
My previous code partner wrote an article on how to achieve [this](https://www.freecodecamp.org/news/how-to-set-unique-interchangeable-index-constraint-in-rails/).

- [x] We haven't written and friendship or user models tests for friendship feature yet. Let's add tests to test associations and all instance method in `user.rb`.

- [x] There is a failing test in your test suite. Let's fix it:
Screenshot from 2019-10-23 21-14-45


#### Please handle these requirements and submit a new code review.

#### Happy Coding :computer:

**Muhammad Ebeid**

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUIRED
Hi Team, good progress on this :+1: , everything works fine the only thing left are test for this milestone, the previous reviewer mentioned it, I guess you guys somehow skipped it. I cannot see test written for this implementation.

Please make changes and request for another review :+1:  

**Raphael Noriode**

**Review by Oghenebrume50 — APPROVED** [review]
### Status: Approved

Good Job Team :+1:  Everything works now, this is approved.

**Raphael Noriode**

**Pull request merged**